src/systems/scene_broadcaster/SceneBroadcaster.cc

```diff
@@ -239,6 +242,16 @@ void SceneBroadcaster::Configure(
   }
 }
 
+//////////////////////////////////////////////////
+void SceneBroadcaster::PreUpdate(const UpdateInfo &_info,
+    EntityComponentManager &_ecm)
+{
+  bool jumpBackInTime = _info.dt < std::chrono::steady_clock::duration::zero();
+  this->dataPtr->changeEvent =  _ecm.HasEntitiesMarkedForRemoval() ||
+        _ecm.HasNewEntities() || _ecm.HasOneTimeComponentChanges() ||
+        jumpBackInTime;
```

Comment:
@nkoenig , I think this comment is still valid: https://github.com/ignitionrobotics/ign-gazebo/pull/497#issuecomment-749839626

The `Has` functions aren't meant to be called during `PreUpdate`. Entities may be modified after this system's `PreUpdate`, and the result you get here becomes invalid.

Reply:
I've tried moving the `scene-broadcaster` plugin to different positions in an sdf world. The component inspector and user commands seem to work for me. Do you have a specific example of the failure case?

Reply:
@chapulina I think you mentioned that you had more information about this?

Reply:
I just took a quick look at this again. I see that the `changeEvent` was moved back to `PostUpdate` on 17bb8f178c5cb97e8c741c5a4cdd10efda90b35a, and that fixes the concern I had. Is the addition of `PreUpdate` still needed?

Reply:
I think removing the preupdate will break setting poses of objects via a service call. See https://github.com/ignitionrobotics/ign-gazebo/pull/497#issuecomment-749650067

Reply:
@chapulina , I confirmed that I need the `PreUpdate` in order to capture change events from service calls.

Reply:
@chapulina , do you have any additional concerns?

Reply:
Ok, I was able to create a scenario where the current implementation fails. Order the systems as follows on the SDF:

1. `SceneBroadcaster`
2. `UserCommands`
3. `Physics`

What happens:

* `PreUpdate`:
    * `SceneBroadcaster` doesn't see any one time changes
    * `UserCommands` creates the `WorldPoseCmd` component
* `Update`:
    * `Physics` removes `WorldPoseCmd`
* `PostUpdate`:
    * `SceneBroadcaster` doesn't see any one time changes, doesn't sent message

I'm thinking of a workaround.

Reply:
Mind taking a look at https://github.com/ignitionrobotics/ign-gazebo/pull/653?